File: src/webfrontend/src/models/Category.ts
import { Comparable } from "./Comparable"

export class Category implements Comparable<Category> {
    public id: number = 0
    public name: string = ""

    constructor(id: number, name: string) {
        this.id = id
        this.name = name
    }
    Equals(other: Category): boolean {
        return this.id === other.id
            && this.name === other.name
    }
}

File: src/Backend.Core/Models/Category.cs
namespace Backend.Models;

public class Category
{
    public int Id { get; set; }
    public string Name { get; set; }
    public int UserId { get; set; }

    public override bool Equals(object? other)
    {
        if (other is not Category otherCat)
        {
            return false;
        }
        return Id == otherCat.Id 
               && Name == otherCat.Name 
               && UserId == otherCat.UserId;
    }
}